src/app/loadout-builder/process/useProcess.ts

```diff
@@ -222,20 +228,27 @@ const groupComparator = chainComparator(
 );
 
 /**
- * This groups items for process depending on whether any general, other or raid mods are locked as follows
- * - If there are general, other or raid mods locked it groups items by (stats, masterworked, modSlot, energyType).
- * - If there are only general mods locked it groupes items by (stats, masterwork, energyType)
- * - If no general, other or raid mods are locked it groups by (stats, masterworked).
+ * To reduce the number of items sent to the web worker we group items by a number of varying
+ * parameters, depending on what mods and armour upgrades are selected.
+ *
+ * After items have been grouped we only send a single item (the first one) as a representitive of
+ * said group. All other grouped items will be available by the swap icon in the UI.
  *
- * Note that assumedMasterwork effects this.
+ * It can group by any number of the following concepts depending on locked mods and armor upgrades,
+ * - Stat distribution
+ * - Masterwork status
+ * - If there are energy requirements for slot independant mods it creates groups split by energy type
+ * - If there are activity mods it will create groups split by specialty socket tag
  */
 function groupItems(
   defs: D2ManifestDefinitions | undefined,
   items: readonly DimItem[],
   statOrder: number[],
   upgradeSpendTier: UpgradeSpendTier,
+  lockItemEnergyType: boolean,
   generalMods: PluggableInventoryItemDefinition[],
-  raidCombatAndLegacyMods: PluggableInventoryItemDefinition[]
+  combatMods: PluggableInventoryItemDefinition[],
+  activityMods: PluggableInventoryItemDefinition[]
```

Comment:
Do "activity mods" cover raid mods? I'm not up to speed on mod naming conventions.

Reply:
Yeah that is a recent change, `activity mods` are raid mods the nightmare mods. I stole the name from somewhere else in the code base, I think @nev-r coined it maybe?

Reply:
yeah the main thing that distinguishes "5th slot" mods is that they are activity-specific. they're sometimes called "interesting" modslots in our codebase (my fault) but `activity` is their mod slot search term, as it seems most accurate for now